src/librustc/diagnostics.rs

```diff
@@ -227,6 +227,23 @@ const Y: i32 = A;
 ```
 "##,
 
+E0014: r##"
+Constant variables' value can only be a constant value or a function. Bad
+example:
+
+```
+const FOO: i32 = { let x = 0; x }; // 'x' isn't a constant nor a function!
+```
+
+To avoid it, you have to replace the non-constant value:
+
+```
+const FOO: i32 = { const X : i32 = 0; X };
+// or even:
+const FOO: i32 = { 0 }; // but brackets are useless here
```

Comment:
It was just in order to stay in the same type of assignation. That's why I added the comment by the way.